src/main/java/Deadline.java

```diff
@@ -0,0 +1,24 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+// a task that need to specify a date to be done before the date
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
```

Comment:
According to the CS2103/T textbook, under the "Guideline: Name well" section in the Code Quality chapter, it says that nouns should be used for classes/variables. Maybe you could consider using a noun to describe your variable. You could maybe use `deadlineDate` or `dateTime` instead of `by`.

Reply:
thanks for reminding. i actually forgot about it